src/getRuntimeChoice.ts

```diff
@@ -0,0 +1,76 @@
+import * as _ from "lodash";
+import * as path from "path";
+import * as clc from "cli-color";
+import * as semver from "semver";
+
+import * as FirebaseError from "./error";
+import * as utils from "./utils";
+
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+var cjson = require("cjson");
+
+export function getRuntimeChoice(sourceDir: string): any {
+  const packageJsonPath = path.join(sourceDir, "package.json");
+  const loaded = cjson.load(packageJsonPath);
+  const choice = loaded.engines;
+  if (!choice) {
+    return null;
+  }
+
+  function nodeVersion(version: string): boolean {
+    return _.isEqual(choice, { node: version });
+  }
+
+  if (nodeVersion("6")) {
+    return _handleNode6();
```

Comment:
I'm not a big fan of relying on function hoisting (defining functions below where you use them), and each of these functions are really quite small. WDYT about something streamlined:

```ts
const ENGINE_RUNTIMES = {
  "6": "nodejs6",
  "8": "nodejs8",
  "10": "nodejs10",
};

const choice = _.get(loaded, "engines.node");
const runtime = ENGINE_RUNTIMES[choice];
if (!runtime) {
  // throw unsupported engine error
}

if (runtime === "nodejs6") {
 // print deprecation warning
}

return runtime
```

This doesn't include the functions SDK check (sorry rushing a bit here) but I think is a little more straightforward as an approach.

Reply:
Agreed - once I removed the warnings from Node 10 there's definitely not much happening in these functions. Will refactor.